### DISCUSSION TITLE
Throwing Error from index.js

### DISCUSSION THREAD

**Discussion by RyanKoech** [reach-sh/reach-lang#805] [Q&A] [answered]
https://github.com/reach-sh/reach-lang/blob/21b070248feb0a345f8ceb1a6ea180853a617103/examples/rps-9-web/index.js#L22

Why am I getting this error?
```Uncaught (in promise) Error: Providers created by environment do not have default addresses. Calling setProviderByEnv or setProviderByName removes this capability. Try removing calls to those functions.```

**Comment by jeapostrophe** [comment, marked as answer]
I'd have to see more of your code and know how you ran it. The error message says what the issue is most of the time. Another possibility is that you are using ETH and don't have Metamask installed.

**Reply by RyanKoech** [reply]
The code is similar, actually an exact copy of the Reach Tutorial code, and I have Metamask installed as well. I also cloned that repo and got the same error.

**Reply by jeapostrophe** [reply]
How did you run the program?

**Reply by RyanKoech** [reply]
`./Reach react`
[Here](https://github.com/RyanKoech/Reach_Tutorial) is the link to the code.

**Reply by jeapostrophe** [reply]
Did you set any environment variables?

**Reply by RyanKoech** [reply]
Only REACH_CONNECTOR_MODE

**Reply by jeapostrophe** [reply]
What did you define it as?

**Reply by RyanKoech** [reply]
ALGO

**Reply by jeapostrophe** [reply]
Ah, so you didn't follow the directions for running with ALGO... https://docs.reach.sh/tut/rps/#p_421

I assume you don't have an ARC11 wallet in your browser

**Reply by RyanKoech** [reply]
Yes! I should have followed on with the instructions. Thanks for your help!
